design/gateway-apis-implementation.md

```diff
@@ -163,37 +169,82 @@ The output of this watcher is Envoy configuration.
 ## Detailed Design
 
 ### Configuration
-Contour will have an entry added to the config file for the Gateway it should watch.
+Contour will have an entry added to the config file for the `ControllerName` it should watch.
 
-This will look like this:
 ```yaml
 gateway:
-  name: gatewayname
-  namespace: gatewaynamespace
+  controllerName: 
+  name: gatewayname  # <--- deprecate
+  namespace: gatewaynamespace  # <-- deprecate
 ```
+Contour will look for a Gateway + GatewayClass that matches the `ControllerName` field specified in the configuration file.
+If it does not find a matching Gateway/GatewayClass, then it will begin processing GatewayAPI resources once they are available. 
 
-The current flag `--experimental-service-apis` will be removed.
-The boolean it represents will be deduced from the following rules:
-- the gateway is configured in the config file.
-- the referenced gateway exists.
+Should two different Gateway or GatewayClasses be found matching the same `ControllerName`, then Contour will utilize the oldest created object and set a status on the other(s) which describes that object as being rejected.
 
-If the gateway is configured but does not exist, then this is a fatal error and Contour will exit.
 If the gateway is removed while Contour is operating, then an error will be logged, and all config associated with the Gateway API will be removed from Envoy.
 
-### Code changes
-Contour already has support for importing the Gateway API objects into its Kubernetes cache.
-However, for some types, we need to be able to keep some details - this design suggests making those details properties of the cache, as `IngressClass` is currently.
+_NOTE: The fields `name` & `namespace` will be deprecated from the configuration file._
+
+#### Managed Envoy
+
+Contour can optionally provision Envoy infrastructure in a Kubernetes cluster.
+A new field will be added to the Contour configuration file which will determine if Contour should dynamically provision a fleet of Envoys with a corresponding service.
+The field will be named `envoy` and contain a single value for now of `managed: true` or `managed: false`.
+
+This field will default to `false` if unspecified.
+
+Example config file:
+```yaml
+envoy:
+  managed: true
+gateway:
+  controllerName: projectcontour.io/ingress-controller
+```
+
+_NOTE: Typically Envoy is deployed as a Daemonset, but not required (could also be a Deployment), so the term `fleet` is used to identify the set of Kubernetes managed pods which represent an Envoy instance._
+
+##### managed: true
+
+If envoy.managed is true, then Contour will automatically create a Envoy fleet/service.
```

Comment:
Given the discussion of Loadbalancer Services below, I'd suggest something like this:
```suggestion
If envoy.managed is true, then Contour will automatically create a Envoy fleet and a load balancer to get traffic to it. See below for a discussion on options for how Contour will do that.
```